### PR TITLE
Add SAPO loss type support for LigerFusedLinearGRPOLoss

### PR DESCRIPTION
**Pull request by yukiu00**
## Summary

* Add **SAPO** (`loss_type="sapo"`) support to **`LigerFusedLinearGRPOLoss`** (chunked loss path)
* Enable TRL's **`GRPOTrainer`** to work with `use_liger_kernel=True` and `loss_type="sapo"`

### Background / Motivation

SAPO (Soft Adaptive Policy Optimization) is a loss variant proposed in the paper ["Soft Advantage Policy Optimization"](https://arxiv.org/abs/2511.20347). It replaces hard clipping with a smooth, temperature-controlled sigmoid-based gate that adaptively attenuates off-policy updates while preserving useful learning signals.

TRL added `loss_type="sapo"` to `GRPOTrainer`, but Liger Kernel did not support it, causing errors when using `use_liger_kernel=True` with `loss_type="sapo"`.

### Changes

**`src/liger_kernel/chunked_loss/grpo_loss.py`**

* Add SAPO loss matching TRL's implementation
  * Sigmoid-based soft gating formula:
    ```python
    sigmoid(τ * (ρ - 1)) * 4 / τ
    ```
  * Support different temperatures for positive/negative advantages (`sapo_temperature_pos`, `sapo_temperature_neg`)
* Use **GRPO-style normalization** for SAPO reduction (per-sequence average, consistent with TRL)
* SAPO does not use clipping metrics (always returns zeros for `is_clipped`)

**`src/liger_kernel/chunked_loss/fused_linear_ppo.py`**

* Add `sapo_temperature_pos` and `sapo_temperature_neg` parameters to base class

**`src/liger_kernel/transformers/grpo_loss.py`**

* Raise explicit error for Triton GRPO loss path (SAPO not supported there)

**`test/chunked_loss/test_grpo_loss.py`**

* Add SAPO to torch reference implementation (`TorchLMHeadGRPO`)
* Add `"sapo"` to parameterized test cases to verify parity with reference

### References

* SAPO Paper: https://huggingface.co/papers/2511.20347
* TRL SAPO implementation: https://github.com/huggingface/trl/blob/1bd2a52ec2d8344050af736d60cdc735181ae4b8/trl/trainer/grpo_trainer.py#L1913

### Related PRs

* #1054 - Add CISPO loss type support for LigerFusedLinearGRPOLoss

## Testing Done

- Hardware Type: RTX3090 24GB
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [ ] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Review by Tcc0403 — APPROVED** [review]
Thank you so much! I simplify the code path for `per_token_loss` calculation. `torch.zeros_like` for `is_clipped` boolean is really a good idea, I applied it to cispo as well.
